citesphere/src/main/java/edu/asu/diging/citesphere/web/user/EditItemController.java

```diff
@@ -25,161 +29,302 @@
+            @PathVariable("itemType") ItemType itemType) {
         List<String> fields = new ArrayList<>();
         citationManager.getItemTypeFields((IUser) authentication.getPrincipal(), itemType)
-            .forEach(f -> fields.add(f.getFilename()));
+                .forEach(f -> fields.add(f.getFilename()));
         return new ResponseEntity<List<String>>(fields, HttpStatus.OK);
     }
-    
-    @RequestMapping(value="/auth/group/{zoteroGroupId}/items/{itemId}/edit", method = RequestMethod.POST)
-    public String storeItem(@ModelAttribute CitationForm form, Authentication authentication, Model model, @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("itemId") String itemId) throws ZoteroConnectionException, GroupDoesNotExistException {
-        ICitation citation = citationManager.getCitation((IUser)authentication.getPrincipal(), zoteroGroupId, itemId);
-        // load authors and editors before detaching
-        citation.getAuthors().forEach(a -> a.getAffiliations().size());
-        citation.getEditors().forEach(e -> e.getAffiliations().size());
-        citation.getOtherCreators().forEach(e -> e.getPerson().getAffiliations().size());
-        citationManager.detachCitation(citation);
-        citationHelper.updateCitation(citation, form);
-        try {
-            citationManager.updateCitation((IUser)authentication.getPrincipal(), zoteroGroupId, citation);
-        } catch (CitationIsOutdatedException e) {
-            citation = citationManager.getCitation((IUser)authentication.getPrincipal(), zoteroGroupId, itemId);
-            ICitation currentCitation = citationManager.getCitationFromZotero((IUser)authentication.getPrincipal(), zoteroGroupId, itemId);
+
+    @RequestMapping(value = "/auth/group/{zoteroGroupId}/items/{itemId}/edit", method = RequestMethod.POST)
+    public String storeItem(@ModelAttribute CitationForm form, Authentication authentication, Model model,
+            @PathVariable("zoteroGroupId") String zoteroGroupId, @PathVariable("itemId") String itemId)
+            throws ZoteroConnectionException, GroupDoesNotExistException {
+        ICitation citation = citationManager.getCitation((IUser) authentication.getPrincipal(), zoteroGroupId, itemId);
+        if (form.getVersion() == citation.getVersion()) {
+
+            // load authors and editors before detaching
+            citation.getAuthors().forEach(a -> a.getAffiliations().size());
+            System.out.println(citation.getAuthors().size());
+            citation.getEditors().forEach(e -> e.getAffiliations().size());
+            citationManager.detachCitation(citation);
+            citationHelper.updateCitation(citation, form);
+
+            try {
+                citationManager.updateCitation((IUser) authentication.getPrincipal(), zoteroGroupId, citation);
+            } catch (CitationIsOutdatedException e) {
+                System.out.println(e);
+            }
```

Comment:
this is not how we handle expceptions

Reply:
also, in this case we want to show the resolve conflict page as we did before

Reply:
The resolve conflict page is being handled by the else condition. An exception here would simply result in a failed update.